src/ocm/hooks/useInfraEnvImageUrl.ts

```diff
@@ -4,35 +4,30 @@ import { Cluster, PresignedUrl } from '../../common';
 import { InfraEnvsAPI } from '../services/apis';
 import { getErrorMessage } from '../../common/utils';
 
+type ImgUrl = {
+  url: PresignedUrl['url'];
+  error: string;
+};
+
 export default function useInfraEnvImageUrl(clusterId: Cluster['id']) {
-  const { infraEnvId, error: infraEnvIdError } = useInfraEnvId(clusterId);
-  const [imageUrl, setImageUrl] = React.useState<PresignedUrl['url']>();
-  const [error, setError] = React.useState('');
+  const { infraEnvId, error: infraEnvError } = useInfraEnvId(clusterId);
 
-  const getImageUrl = React.useCallback(async () => {
+  const getImageUrl = React.useCallback(async (): Promise<ImgUrl> => {
     try {
       if (!infraEnvId) {
-        return;
+        return { url: '', error: infraEnvError || 'Missing infraEnv' };
```

Comment:
```suggestion
        return { url: '', error: infraEnvError || 'Missing infraEnv' };
```
can we omit the `url` prop?

Reply:
Since we want to return two different things, we need to wrap the response payload in an object.
If we just return an string, we wouldn't know if it's an error or the URL.